### PR TITLE
Add pipeline for tsp nightly build

### LINKED ISSUE #3299: Add nightly CI to test latest TSP releases
The TSP team push to npm a package for every single merge to main. The objective of this issue is that the language repo has a nightly build that:
- Install the latest "next" release from npm for all dependencies (compile, azure-core, etc.)
- Run the testsuite as usual

Implementation detail: using [ncu](https://www.npmjs.com/package/npm-check-updates) with `--target @next` should allow for simple package.json update to latest dev version (not tested at the time of this issue).

ETA should be before the next release of the TSP team (early May)

### PR DESCRIPTION
**Pull request by pshao25**
Fix #3299 

Sample build: https://dev.azure.com/azure-sdk/internal/_build/results?buildId=2793719&view=results

### CODE REVIEW

File: eng/CheckInstalledDevTsp.ps1
Comment (lirenhe) [inline eng/CheckInstalledDevTsp.ps1:11]: Write-Error

File: eng/CheckInstalledDevTsp.ps1
Comment (lirenhe) [inline eng/CheckInstalledDevTsp.ps1:17]:
```suggestion
    Write-Host "The compiler should not be installed in the emitter folder."
```

File: eng/Generate.ps1
Comment (chunyu3) [inline eng/Generate.ps1:73]: Is `--config` a parameter of typespec compiler after 0.45.*? If so, this parameter is not recognized by 0.44, will it break our current pipeline for 0.44? Please make sure it does not break current pipeline.

**Review by chunyu3 — APPROVED** [review]

**Review by lirenhe — APPROVED** [review]

File: eng/Generate.ps1
Comment (pshao25) [reply eng/Generate.ps1:73]: No, I have tested.

**Review by m-nash — CHANGES_REQUESTED** [review]
I think we are going to go a slightly different route here so lets hold off on this PR.

**Comment by pshao25** [comment]
> I think we are going to go a slightly different route here so lets hold off on this PR.

@m-nash What different route are you mentioning?

File: eng/pipelines/tsp-nightly.yml
Comment (pshao25) [inline eng/pipelines/tsp-nightly.yml:1]: Change this before merge.
```
schedules:
- cron: '0 0 * * *'
  displayName: Daily midnight build for tsp
  branches:
    include:
    - feature/v3
```

**Comment by m-nash** [comment]
> > I think we are going to go a slightly different route here so lets hold off on this PR.
> 
> @m-nash What different route are you mentioning?

The route we are going here is the one we discussed and have been working on for a couple months now which is to utilize our cross repo interop.

**Comment by m-nash** [comment]
Closing since the other pipeline is now merged

**Pull request closed without merging**
